source/fe/mapping_q_generic.cc

```diff
@@ -1140,6 +1114,42 @@ MappingQGeneric<dim,spacedim>::get_degree() const
 
 
 template <int dim, int spacedim>
+const std::vector<Table<2,double> > &
+MappingQGeneric<dim,spacedim>::get_support_point_weights_perimeter_to_interior() const
+{
+  Assert(dim > 0 && dim <= 3, ExcImpossibleInDim(dim));
+  Assert(polynomial_degree > 1,
+         ExcInternalError("This should not be called for degree=1"));
+
+  // initialization upon first request
+  if (support_point_weights_perimeter_to_interior.empty())
+    {
+      Threads::Mutex::ScopedLock lock(this->mutex);
+
+      std::vector<Table<2,double> > output(dim);
+      // fill the 1D interior weights
+      QGaussLobatto<1> quadrature(polynomial_degree+1);
+      output[0].reinit(polynomial_degree-1, GeometryInfo<1>::vertices_per_cell);
+      for (unsigned int q=0; q<polynomial_degree-1; ++q)
+        for (unsigned int i=0; i<GeometryInfo<1>::vertices_per_cell; ++i)
+          output[0](q,i) = GeometryInfo<1>::d_linear_shape_function(quadrature.point(q+1),
+                                                                    i);
+
+      if (dim > 1)
+        output[1] = compute_support_point_weights_on_quad(polynomial_degree);
+
+      if (dim > 2)
+        output[2] = compute_support_point_weights_on_hex(polynomial_degree);
+
+      const_cast<std::vector<Table<2,double> > &>(support_point_weights_perimeter_to_interior)
+      .swap(output);
```

Comment:
Instead of using swap, I think the more modern way of doing this is to use dst = std::move(src).